Content.Server/Body/Systems/BloodstreamSystem.cs

```diff
@@ -159,18 +161,26 @@ private void OnDamageChanged(EntityUid uid, BloodstreamComponent component, Dama
         if (bloodloss.Empty)
             return;
 
+        // Does the calculation of how much bleed rate should be added/removed, then applies it
         var oldBleedAmount = component.BleedAmount;
         var total = bloodloss.Total;
         var totalFloat = total.Float();
         TryModifyBleedAmount(uid, totalFloat, component);
 
-        var prob = Math.Clamp(totalFloat / 50, 0, 1);
+        /// <summary>
+        ///     Critical hit. Causes target to lose blood, using the bleed rate modifier of the weapon, currently divided by 5
+        ///     The crit chance is currently the bleed rate modifier divided by 25.
+        ///     Higher damage weapons have a higher chance to crit!
+        /// </summary>
+        var prob = Math.Clamp(totalFloat / 25, 0, 1);
         var healPopupProb = Math.Clamp(Math.Abs(totalFloat) / 25, 0, 1);
         if (totalFloat > 0 && _robustRandom.Prob(prob))
         {
             TryModifyBloodLevel(uid, (-total) / 5, component);
             _audio.PlayPvs(component.InstantBloodSound, uid);
         }
+
+        // Heat damage will always cauterize, this message simply has a chance of showing up as feedback to avoid spam.
         else if (totalFloat < 0 && oldBleedAmount > 0 && _robustRandom.Prob(healPopupProb))
```

Comment:
This has a chance of not giving any feedback at all, or giving too much feedback.
Just add an accumulator to the component or something to stop popup spam.

[also this](https://cdn.discordapp.com/attachments/310555209753690112/1001323526223052881/ss13_code_shit.png)

Reply:
It's just the feedback fortunately... while less of an issue it's still absolutely an issue, and not one I know how to fix...